### PR TITLE
Rhino brep trimmed

### PR DESCRIPTION
**Pull request by chenkasirer**
* `RhinoBrep.trimmed` returns now a single brep as a result instead of a list of breps. If no result available a `BrepTrimmingError` is raised.

This fixes a discrepancy between the `RhinoBrep` implementation and the documented behavior. 

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [x] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by jf---** [comment]
@gonzalocasas || @tomvanmele this looks good, only thing that comes to mind if handling precision is following the new `compas.precision` idiom? 

The description is not super clear -- @chenkasirer this is fusing a nr of `faces`, right?
In that sense the method is a little tricky to understand, and if topological changes (if faces are fused) are invoked, a little more on the intent of the method is needed I think. fused faces when invoking `trimmed` is semantically not crystal clear I feel.

**Review by tomvanmele — DISMISSED** [review]
LGTM

**Comment by tomvanmele** [comment]
actually, sorry i take that back, tolerance handling is indeed wrong...

File: src/compas_rhino/geometry/brep/brep.py [hunk 1]
```diff
@@ -394,26 +404,38 @@ def trimmed(self, plane, tolerance=TOLERANCE):
...
-        for result in results:
-            capped = result.CapPlanarHoles(TOLERANCE)
-            if capped:
-                result = capped
-            breps.append(RhinoBrep.from_native(result))
-        return breps
+        if not results:
+            raise BrepTrimmingError("Trim operation ended with no result")
+        result = results[0]
+        capped = result.CapPlanarHoles(TOLERANCE)
```
Comment (tomvanmele) [inline src/compas_rhino/geometry/brep/brep.py]: in the signature, `tolerance=None`. then `tolerance = tolerance or TOL.absolute`, with at the top (if it is not already there) `from compas.tolerance import TOL`

Comment (tomvanmele) [reply src/compas_rhino/geometry/brep/brep.py]: and then, finally, `capped = result.CapPlanarHoles(tolerance)`

**Review by tomvanmele — CHANGES_REQUESTED** [review]
use of tolerance should be updated

**Comment by tomvanmele** [comment]
@chenkasirer ping

File: src/compas_rhino/geometry/brep/brep.py (hunk 1 above)
Comment (chenkasirer) [reply src/compas_rhino/geometry/brep/brep.py]: @tomvanmele done.
Just curious why not `tolerance=TOL.absolute` in signature? as it returns a `float` (immutable)

**Comment by chenkasirer** [comment]
> @gonzalocasas || @tomvanmele this looks good, only thing that comes to mind if handling precision is following the new `compas.precision` idiom?
> 
> The description is not super clear -- @chenkasirer this is fusing a nr of `faces`, right? In that sense the method is a little tricky to understand, and if topological changes (if faces are fused) are invoked, a little more on the intent of the method is needed I think. fused faces when invoking `trimmed` is semantically not crystal clear I feel.

Hey @jf--- !
Not sure what you mean by fusing faces here, could you explain? `Brep.trimmed` just returns all the results of a trim operation. trimming with plane would then typically result in two breps (one on each side of the plane), but maybe I'm missing something.

**Review by tomvanmele — APPROVED** [review]
LGTM

File: src/compas_rhino/geometry/brep/brep.py (hunk 1 above)
Comment (tomvanmele) [reply src/compas_rhino/geometry/brep/brep.py]: that would have also been an option indeed. as long as it is not a hard coded tolerance value...

**Pull request merged**